src/OpenTK.Windowing.Desktop/Win32WindowProc.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using OpenTK.Windowing.GraphicsLibraryFramework;
+
+namespace OpenTK.Windowing.Desktop
+{
+    /// <summary>
+    /// Windows-specific Win32 message-handling.
+    /// </summary>
+    internal unsafe class Win32WindowProc : IDisposable
+    {
+        [DllImport("user32.dll", EntryPoint = "SetWindowLong")]
+        private static extern int SetWindowLong32(IntPtr hWnd, int nIndex, int dwNewLong);
```

Comment:
Short version: that API call isn't available to 32-bit builds:

![image](https://github.com/opentk/opentk/assets/794270/289d8c32-fc48-4a0f-bf58-677cd6a27682)

The docs for [`SetWindowLongPtrW`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowlongptrw) indicates it isn't appropriate for 32-bit. Their guidance is to use the macro, which isn't an option, of course, and they probably could have stated this more clearly, but:

> Note  To write code that is compatible with both 32-bit and 64-bit versions of Windows, use SetWindowLongPtr. When compiling for 32-bit Windows, SetWindowLongPtr is defined as a call to the [SetWindowLong](https://learn.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-setwindowlonga) function.

Just in case I was missing something, this Q&A on [StackOverflow](https://stackoverflow.com/questions/3343724/) is pretty old, but both Hans Passant _and_ Stephen Cleary both suggest what I have done, and my 32-bit / 64-bit resolver is similar to how it's done in .NET WinForms (checking `IntPtr.Size`). If only we could get Jon Skeet to weigh in, we'd have some sort of Holy Geek Trinity of consensus!

Also, I'm not sure `SetWindowLongPtrW` automatically converts a 64-bit OS pointer for a 32-bit application, which is the purpose of the `dwNewLong.ToInt32()` in my code, which is something I wrote in a very old .NET Framework library which supported both 32-bit and 64-bit.

Reply:
Named the resolver function `SetWindowPointer` and imported the Win32 calls with their correct names.